source/features/unfinished-comments.tsx

```diff
@@ -1,26 +1,34 @@
 import select from 'select-dom';
+import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
 let documentTitle: string | undefined;
 
 function hasDraftComments(): boolean {
-	return select.all('textarea').some(textarea => textarea.value.length > 0 && textarea.offsetWidth > 0);
+	return select.all('textarea').some(textarea => textarea.value.length > 0 && textarea.offsetWidth > 0 && !textarea.closest('.rgh-is-sending-comment'));
 }
 
-async function updateDocumentTitle(): Promise<void> {
-	if (documentTitle) {
-		document.title = documentTitle;
-		documentTitle = undefined;
-	} else if (document.visibilityState === 'hidden' && hasDraftComments()) {
+function updateDocumentTitle(): void {
+	if (document.visibilityState === 'hidden' && hasDraftComments()) {
 		documentTitle = document.title;
 		document.title = '(Draft comment) ' + document.title;
+	} else if (documentTitle) {
+		document.title = documentTitle;
+		documentTitle = undefined;
 	}
+
+	select('.rgh-is-sending-comment')?.classList.remove('rgh-is-sending-comment');
```

Comment:
Unfortunately this also isn't compatible with `wait-for-ci` if the user switches to the tab twice (because the first switch will remove the class)

Reply:
```suggestion
```